app/models/abstract_contributor.rb

```diff
@@ -63,6 +63,13 @@ class AbstractContributor < ApplicationRecord
   validates :orcid, format: { with: Orcid::REGEX }, allow_nil: true, if: :person?
   validates :orcid, absence: true, unless: :person?
 
+  # see https://github.com/sul-dlss/happy-heron/issues/2240
+  # Sometimes a contributor is saved before the work_version it belongs to, causing a
+  # null constraint violation in the database.  This forces the work_version to be saved first.
+  before_create do
```

Comment:
I believe that `belongs_to :work_version` already does this. `belongs_to `is required by default.  You can provide `optional: true`, if you allow nil.

Reply:
Ok, I figured as much -- out of other ideas though.